core/src/main/java/org/mule/runtime/core/internal/event/EventInternalContextResolver.java

```diff
@@ -49,6 +50,6 @@ public CTX getCurrentContextFromEvent(CoreEvent event) {
   }
 
   public CoreEvent eventWithContext(CoreEvent event, CTX context) {
-    return quickCopy(event, singletonMap(internalParametersKey, context));
+    return quickCopy(event, of(internalParametersKey, context));
```

Comment:
the singletonMap is already immutable, i wouldn't update those

Reply:
we don't need these being immutable. Doing it like this is faster.